### DISCUSSION TITLE
connectHits regression over allItems

### DISCUSSION THREAD

**Discussion by dianos** [algolia/instantsearch#4915] [General]
I'm wondering how to use mustache templates with connectHits ?  **allItems** allowed a very easy way to upgrade an **item** template to have control over the whole list.  And it was discontinued in favor of a possibly more customizable but much more convoluted approach.  Why can't the two exist side by side?

**Comment by Haroenv** [comment]
Could you give more information to what your use case for allItems is? You're right that it enabled some use cases that are now done in a different way. We made this decision as we saw many people struggle with the hogan/mustache syntax for anything harder than just looping, and that kind of logic fit better in either full custom dom in a connector or changing the list in transformItems.

To help you migrate now, could you explain what you're trying to achieve?

**Comment by dianos** [comment]
my use case is pretty simple: I wanted to replace the ol/li structure with dynamic bootstrap driven layout that adjusts the amount of items in a row depending on screen size, and what is actually displayed for each item. allItems allowed me to keep the same template code with little addition: 
```
 allItems: `<div rowstyles>
                                {{#hits}}
                                <div itemstyles>
                                   large mustache template
                                </div>
                                {{/hits}}
                            </div>`
```
I understand the need for connectHits. I just don't understand why the allItems template was removed.   It might provide a subset of features but it provides them almost instantly.

**Reply by Haroenv** [reply]
I see how that can be simpler, but technically it was causing issues and inconsistencies with the other widgets, which is why in https://github.com/algolia/instantsearch.js/pull/3096 allItems was removed. you can still use innerHTML providing you sanitize input in the connector
